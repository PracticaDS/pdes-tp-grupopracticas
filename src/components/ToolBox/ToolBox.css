.ToolBox{
    height: 350px;
    width: 30%;
    justify-self: end;
}


div.ButtonToolbox.seleccionado img {
  -webkit-filter: brightness(110%) saturate(160%) sepia(30%) hue-rotate(350deg);
  filter: brightness(110%) saturate(160%) sepia(100%) hue-rotate(350deg);
}

h3 {
    color:orange;
}

table {
    background-color: white;
    border-radius: 2%;
    margin: 0 auto;
}