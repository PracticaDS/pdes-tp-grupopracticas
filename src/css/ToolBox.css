.ToolBox{
    height: 350px;
    width: 30%;
    align-content: center;
}


div.ButtonToolbox.seleccionado img {
    position: relative;
    -webkit-filter: brightness(110%) saturate(160%) sepia(30%) hue-rotate(350deg);
    filter: brightness(110%) saturate(160%) sepia(100%) hue-rotate(350deg);
  }
  div.ButtonToolbox.seleccionado img::before {
    content: "";
    display: block;
    height: 100%;
    width: 100%;
    top: 0;
    left: 0;
    position: absolute;
    pointer-events: none;
    mix-blend-mode: screen;
    opacity: 0.3;
    background: rgba(204, 68, 0, 1);
  }

h3{
    color:orange;
}

table{
    background-color: white;
    border-radius: 10%;
    margin: 0 auto;
}