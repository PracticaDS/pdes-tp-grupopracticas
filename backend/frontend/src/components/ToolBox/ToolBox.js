import React, { Component } from 'react';
import ButtonToolBox from '../ButtonToolBox/ButtonToolBox';
import './ToolBox.css';

import boton_furnace     from '../../images/furnace.PNG';
import boton_seller      from '../../images/seller.PNG';
import boton_crafter     from '../../images/crafter.PNG';
import boton_start       from '../../images/starter.PNG';
import boton_transporter from '../../images/transporter.PNG';

import boton_move   from '../../images/move.PNG';
import boton_borrar from '../../images/delete.PNG';
import boton_rotar  from '../../images/rotate.PNG';
import EditionButton from '../EditionButton/EditionButton';

import { DELETE, MOVE, ROTATE } from '../../utils/EditionTypes'


export default class ToolBox extends Component {

   constructor(props) {
    super(props);
    this.state = { toolSeleccionado: "" };
  }

  crearBoton(machine){
    return <ButtonToolBox 
              seleccionado={machine.name === this.state.toolSeleccionado}
              name={machine.name} 
              src={machine.src} 
              price={machine.price}
            /> 
  }

  createEditionButton(props){
    return <EditionButton 
              seleccionado={props.name === this.state.toolSeleccionado}
              machine={props.name} 
              src={props.src} 
              type={props.type}
            /> 
  }

  render() {
    return (
      <div className="ToolBox">
        <h3>Máquinas</h3>
          <table>
            <tbody>
              <tr>
                <td>{this.crearBoton({name: "starter", src: boton_start, price: 400 })}</td>
                <td>{this.crearBoton({name: "crafter", src: boton_crafter, price: 500 })}</td>
              </tr>
              <tr>
                <td>{this.crearBoton({name: "seller", src: boton_seller, price: 300 })}</td>
                <td>{this.crearBoton({name: "furnace", src: boton_furnace, price: 200 })}</td>
              </tr>
              <tr>
                <td>{this.crearBoton({name: "transporter", src: boton_transporter, price: 100 })}</td>
              </tr>
            </tbody>
          </table>

          <h3>Edición</h3>
          <table>
            <tbody>
              <tr>
                <td>{this.createEditionButton({name: "delete", src: boton_borrar, type: DELETE })}</td>
                <td>{this.createEditionButton({name: "move", src: boton_move, type: MOVE})}</td>
              </tr>
              <tr>
                <td>{this.createEditionButton({name: "rotate", src: boton_rotar, type: ROTATE })}</td>
              </tr>
            </tbody>
          </table>
      </div>
    )
  }
}