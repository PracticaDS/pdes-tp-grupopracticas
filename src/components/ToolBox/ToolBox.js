import React, { Component } from 'react';
import ButtonToolBox from '../ButtonToolBox/ButtonToolBox';
import './ToolBox.css';

import boton_furnace     from '../../images/furnace.PNG';
import boton_seller      from '../../images/seller.PNG';
import boton_crafter     from '../../images/crafter.PNG';
import boton_start       from '../../images/starter.PNG';
import boton_transporter from '../../images/transporter.PNG';

import boton_move   from '../../images/move.PNG';
import boton_borrar from '../../images/delete.PNG';
import boton_rotar  from '../../images/rotate.PNG';
import EditionButton from '../EditionButton/EditionButton';

import { DELETE, MOVE, ROTATE } from '../../utils/EditionTypes'


export default class ToolBox extends Component {

   constructor(props) {
    super(props);
    this.state = { toolSeleccionado: "" };
  }

  crearBoton(props){
    return <ButtonToolBox 
              seleccionado={props.nombre === this.state.toolSeleccionado}
              nombre={props.nombre} 
              src={props.src} 
            /> 
  }

  createEditionButton(props){
    return <EditionButton 
              seleccionado={props.nombre === this.state.toolSeleccionado}
              nombre={props.nombre} 
              src={props.src} 
              type={props.type}
            /> 
  }

  render() {
    return (
      <div className="ToolBox">
              <h3>Máquinas</h3>
                <table>
                  <tbody>
                    <tr>
                      <td>{this.crearBoton({nombre: "starter", src: boton_start })}</td>
                      <td>{this.crearBoton({nombre: "crafter", src: boton_crafter })}</td>
                    </tr>
                    <tr>
                      <td>{this.crearBoton({nombre: "seller", src: boton_seller })}</td>
                      <td>{this.crearBoton({nombre: "furnace", src: boton_furnace })}</td>
                    </tr>
                    <tr>
                      <td>{this.crearBoton({nombre: "transporter", src: boton_transporter })}</td>
                    </tr>
                  </tbody>
                </table>

                <h3>Edición</h3>
                <table>
                  <tbody>
                    <tr>
                      <td>{this.createEditionButton({nombre: "delete", src: boton_borrar, type: DELETE })}</td>
                      <td>{this.createEditionButton({nombre: "move",   src: boton_move, type: MOVE})}</td>
                    </tr>
                    <tr>
                      <td>{this.createEditionButton({nombre: "rotate",   src: boton_rotar, type: ROTATE })}</td>
                    </tr>
                  </tbody>
                </table>
      </div>
    )
  }
}