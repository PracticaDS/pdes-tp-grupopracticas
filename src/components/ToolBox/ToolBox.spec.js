import ToolBox from './ToolBox'
import React from 'react';
import ReactDOM from 'react-dom';
import ToolBox from './ToolBox'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ToolBox />, div);
  ReactDOM.unmountComponentAtNode(div);
});