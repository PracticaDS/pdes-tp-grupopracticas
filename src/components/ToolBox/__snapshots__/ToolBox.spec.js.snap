// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolBox should render correctly whithout props 1`] = `
<div
  className="ToolBox"
>
  <h3>
    Máquinas
  </h3>
  <table>
    <tbody>
      <tr>
        <td>
          <Connect(ButtonToolBox)
            nombre="starter"
            seleccionado={false}
            src="starter.PNG"
          />
        </td>
        <td>
          <Connect(ButtonToolBox)
            nombre="crafter"
            seleccionado={false}
            src="crafter.PNG"
          />
        </td>
      </tr>
      <tr>
        <td>
          <Connect(ButtonToolBox)
            nombre="seller"
            seleccionado={false}
            src="seller.PNG"
          />
        </td>
        <td>
          <Connect(ButtonToolBox)
            nombre="furnace"
            seleccionado={false}
            src="furnace.PNG"
          />
        </td>
      </tr>
      <tr>
        <td>
          <Connect(ButtonToolBox)
            nombre="transporter"
            seleccionado={false}
            src="transporter.PNG"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <h3>
    Edición
  </h3>
  <table>
    <tbody>
      <tr>
        <td>
          <Connect(EditionButton)
            nombre="delete"
            seleccionado={false}
            src="delete.PNG"
            type="DELETE"
          />
        </td>
        <td>
          <Connect(EditionButton)
            nombre="move"
            seleccionado={false}
            src="move.PNG"
            type="MOVE"
          />
        </td>
      </tr>
      <tr>
        <td>
          <Connect(EditionButton)
            nombre="rotate"
            seleccionado={false}
            src="rotate.PNG"
            type="ROTATE"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
